.github/workflows/dispatch-yocto-build-deploy.yml

```diff
@@ -0,0 +1,124 @@
+name: "Test Yocto"
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+  pull_request_target:
+    types: [opened, synchronize]
+    branches:
+      - "main"
+      - "master"
+
+  workflow_dispatch:
+    # you may only define up to 10 `inputs` for a `workflow_dispatch` event
+    inputs:
+      device-repo:
+        description: balenaOS device repository (owner/repo)
+        required: true
+        type: string
+        default: balena-os/balena-generic
+      device-repo-ref:
+        description: balenaOS device repository tag, branch, or commit to build
+        required: false
+        type: string
+        default: master
+      meta-balena-ref:
+        description: meta-balena ref if not the currently pinned version
+        required: false
+        type: string
+      yocto-scripts-ref:
+        description: balena-yocto-scripts ref if not the currently pinned version
+        required: false
+        type: string
+      machine:
+        description: yocto board name
+        required: true
+        type: string
+        default: generic-amd64
+      environment:
+        description: Select deploy environment
+        required: false
+        type: choice
+        options:
+          - >
+            {
+              "environment": "balena-staging.com",
+              "s3-bucket": "resin-staging-img",
+              "s3-region": "us-east-1",
+              "aws-subnet": "subnet-0d73c1f0da85add17",
+              "aws-security-group": "sg-09dd285d11b681946"
+            }
+          - >
+            {
+              "environment": "balena-cloud.com",
+              "s3-bucket": "resin-production-img-cloudformation",
+              "s3-region": "us-east-1",
+              "aws-subnet": "subnet-02d18a08ea4058574",
+              "aws-security-group": "sg-057937f4d89d9d51c"
+            }
+      deploy-s3:
```

Comment:
How do these interact with the global `deploy`? Could we make it to have just the `deploy` yes/no switch, and avoid all these s3/hostapp/ami switches?

Reply:
The issue I see is that AMI is only deployed for a couple device types so it would be easy to set to true at the device-repo level when calling the shared workflow, and leave as false for everything else. At least until we can get the AMI from a source of truth like contracts. I'll add a TODO.

Reply:
Also being able to skip S3 and HostApp deploys is useful for testing at this stage. We may not always expose them.